docs/StateTransitionLogger.md

```diff
@@ -0,0 +1,83 @@
+# Kōjō State Transition Logger
+* Start Date: 2019-08-08
+* Author: Przemyslaw Mucha (przemyslaw.mucha@55places.com)
+
+## Summary
+This document proposes a design for Kōjō logic that will emit log messages for every state transition of a job (e.g. from `waiting` to `working`)
+
+## Problem
+* Why are we doing this?
+    * To gain visibility into the lifecycle of any given job
+* What use cases does it support?
+    * Forensic analysis of a job that didn't behave as expected
+    * Monitoring how much work is being done (more accurately than polling `kojo_job`)
+    * Monitoring how much time is spent in each state
+
+## Proposed Solution
+Job state transitions are persisted to RDBMS in `Neighborhoods\Kojo\State\Service::applyRequest()`.
+To that logic we will add another `INSERT` into a new table: `kojo_job_state_transitions` (working title).
+The schema of that table will include all job information (ID, type, etc.) as well as process information (execution environment, PID, etc.).
+Those two statements will be wrapped in a database transaction to ensure that nothing is written to `kojo_job_state_transitions` unless the actual transition succeeds.
```

Comment:
@mucha55 how are you planning on using PDO to accomplish this with user-land transactions? testing [PDO::inTransaction](https://www.php.net/manual/en/pdo.intransaction.php)?

Reply:
Doctrine DBAL supports "nested" transactions, but even without them this is fine: If there's a userspace transaction happening, because it being rolled back will mean the job state never changes, which means there shouldn't be anything written to `kojo_job_state_transitions` anyway

Reply:
Per an offline discussion, we have to use `\PDO` for managing the transaction, since a Doctrine Connection isn't aware of what happens in the `\PDO` connection that it was instantiated with